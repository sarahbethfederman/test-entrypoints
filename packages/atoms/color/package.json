{
  "name": "@auscred/color",
  "version": "0.0.1",
  "source": "src/Color.tsx",
  "main": "dist/Colour.js",
  "types": "dist/Colour.d.js",
  "files": [
    "lib"
  ],
  "keywords": [],
  "author": "Lendi Devs",
  "license": "ISC",
  "peerDependencies": {
    "@types/react": "^16.4.0",
    "react": "^16.4.0",
    "styled-components": "^3.3.3"
  },
  "scripts": {
    "build": "tsc --pretty"
  }
}
  