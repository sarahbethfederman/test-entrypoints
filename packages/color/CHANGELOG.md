2.0.2 - New Changelog
