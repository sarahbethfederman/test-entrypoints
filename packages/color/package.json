{
  "name": "@lendi-ui/color",
  "version": "3.0.0",
  "description": "Color utilities for color.",
  "source": "src/index.ts",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/types/index.d.ts",
  "files": [
    "dist"
  ],
  "license": "ISC",
  "peerDependencies": {
    "@lendi-ui/theme": "^6.0.0",
    "styled-components": "^3"
  },
  "dependencies": {
    "@lendi-ui/breakpoint": "^2.0.0"
  },
  "scripts": {
    "build:esm": "tsc --project tsconfig.esm.json",
    "build:types": "tsc --project tsconfig.types.json",
    "build": "yarn run build:types && yarn run build:esm"
  }
}
