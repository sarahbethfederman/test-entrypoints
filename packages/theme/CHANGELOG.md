## [9.1.0] 2020-06-29 [FUNNEL-1473](https://creditandfinance.atlassian.net/browse/FUNNEL-1473)

## 10.0.0-beta.0

### Minor Changes

- preconstruct infra update

### Patch Changes

- Updated dependencies [undefined]
  - @lendi-ui/breakpoint@6.0.0-beta.0
- Change `--lend-ui-size` to use new t-shirt breakpoints.

## [9.0.0] 2020-06-29

### Changed

- lendi-ui-size root size breakpoint changed to 576px instead of 577px

## [8.1.0] 2019-10-02 [HUB-589](https://creditandfinance.atlassian.net/browse/HUB-589)

### Changed

- Enhancement LUI theme with default font-family for body and font-weight for heading.

## [8.0.1] 2019-9-29 [HUB-589](https://creditandfinance.atlassian.net/browse/HUB-589)

### Changed

- added font family defaults based on theme

## [8.0.0] 2019-9-23 - 2019-08-15 - [HUB-412](https://creditandfinance.atlassian.net/browse/HUB-412)

## [7.0.0] 2019-9-23 - 2019-08-15 - [HUB-412](https://creditandfinance.atlassian.net/browse/HUB-412)

### Changed

- Color is now a type
- ColorPrefix is now a type
- theme.color object has been flattened
- Update typescript to version 3.5.2 and react to version 16.8.4

## [6.2.0] - 2019-05-17 - [HUB-174](https://creditandfinance.atlassian.net/browse/HUB-174)

### Changed

- Module format reverted back to CJS.

### Deleted

- Dropped "module" property from package.json.

## [6.1.1] - 2019-05-10 - [HUB-137](https://creditandfinance.atlassian.net/browse/HUB-137)

### Added

- Add `-ms-clear` into theme injectGlobal to prevent cross icon in input by Edge and IE browser default style

## [6.1.0] - 2019-05-02 - [HUB-88](https://creditandfinance.atlassian.net/browse/HUB-88)

### Changed

- replace `border-radius: 6px` with select function in theme

## [6.0.2] - 2019-05-09 - [HUB-92](https://creditandfinance.atlassian.net/browse/HUB-92)

### Changed

- Updated CHANGELOG to match required parsing format.

## [6.0.1] - 2019-05-08 - [HUB-54](https://creditandfinance.atlassian.net/browse/HUB-54)

### Changed

- Change module main entry point to esm in package.json

## [6.0.0] - 2019-04-30 - [HUB-54](https://creditandfinance.atlassian.net/browse/HUB-54)

### Deleted

- Remove the CJS bundles.

## [5.0.0] - 2019-03-06 - [BRAVO-387](https://creditandfinance.atlassian.net/browse/BRAVO-387)

### Changed

- Added optional theme kind prop.

## [4.0.2] - 2019-02-25 - [BRAVO-511](https://creditandfinance.atlassian.net/browse/BRAVO-511)

### Changed

- Change warning colour to meet accessibility standards.

## [4.0.1]

### Added

- Added CHANGELOG file.
