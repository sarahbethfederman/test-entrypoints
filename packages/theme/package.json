{
  "name": "@lendi-ui/theme",
  "version": "9.1.0",
  "source": "src/index.ts",
  "main": "dist/cjs/index.js",
  "types": "dist/types/index.d.ts",
  "files": [
    "dist"
  ],
  "keywords": [],
  "author": "Lendi Devs",
  "license": "ISC",
  "peerDependencies": {
    "@lendi-ui/breakpoint": "^5.0.0",
    "@types/react": "^16.8",
    "@types/styled-components": "^4.1.18",
    "react": "^16.8",
    "styled-components": "^4"
  },
  "scripts": {
    "build:cjs": "tsc --project tsconfig.cjs.json",
    "build:types": "tsc --project tsconfig.types.json",
    "build": "yarn run build:types && yarn run build:cjs"
  }
}
