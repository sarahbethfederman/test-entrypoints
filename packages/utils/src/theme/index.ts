export type ColorKeys = '25' | '50' | '100' | '200' | '300' | '400' | '500' | '600' | '700' | '800' | '900' | '950';

export type ColorsMap = { [key in ColorKeys]: string };

export type Typography = {
  heading: {
    fontFamily: string;
  };

  body: {
    fontFamily: string;
  };
};

export type Logo = {
  logoName: string;
};

export type Colors = {
  primary: ColorsMap;
  secondary: ColorsMap;
  tertiary: ColorsMap;
  shade: ColorsMap;
  info: ColorsMap;
  warn: ColorsMap;
  error: ColorsMap;
  success: ColorsMap;
};

export type ThemeMap = {
  typography: Typography;
  colors: Colors;
  logo: Logo;
};
