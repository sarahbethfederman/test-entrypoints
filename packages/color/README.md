# @lendi-ui/color

Color style with `bg`(background color), `fg`(font color) and `color`(every CSS property relate to color).

## Installation

```
yarn add @lendi-ui/color
```

## Usage

- `bg` for styling background-color.

```jsx
import { bg } from '@lendi-ui/color';
const Square = styled.div`
  width: 2em;
  height: 2em;
  ${bg('warn.500')};
`;
```

- `fg` for styling font-color.

```jsx
import { fg } from '@lendi-ui/color';
const WarningText = styled.span`
  ${fg('warn.500')};
`;
```

- `color` for styling every CSS porperty relate to color.

```jsx
import { color } from '@lendi-ui/color';
const Circle = styled.div`
  width: 0;
  border-radius: 1em;
  border: 1em solid ${color('primary.500')};
`;
```
