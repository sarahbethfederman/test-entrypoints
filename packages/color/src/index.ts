import { css } from 'styled-components';
import { select } from '@lendi-ui/theme';
import { map, BreakpointValue, BreakpointValueMap } from '@lendi-ui/breakpoint';

export type NameOrNameMap = BreakpointValue<string> | BreakpointValueMap<string>;

export function color(name: NameOrNameMap) {
  return select(`colors.${name}`);
}

export function fg(name: NameOrNameMap) {
  return map(
    name,
    (n?: string) =>
      n === undefined
        ? ''
        : css`
            color: ${color(n)};
          `
  );
}

export function bg(name: NameOrNameMap) {
  return map(
    name,
    (n?: string) =>
      n === undefined
        ? ''
        : css`
            background-color: ${color(n)};
          `
  );
}
