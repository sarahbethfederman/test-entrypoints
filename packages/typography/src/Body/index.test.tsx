import * as React from 'react';
import { shallow, mount } from 'enzyme';

import { theme } from '@lendi-ui/theme';
import { color as getColor } from '@lendi-ui/color';
import { Body, BodySize, BodyAlign } from '..';

describe('Body', () => {
  const sizes: BodySize[] = [undefined, 'lg', 'md', 'sm'];
  const colors = [undefined, 'primary.500', 'shade.0', 'shade.1000'];
  const alignments: BodyAlign[] = [undefined, 'left', 'center', 'right', { mobile: 'left', desktop: 'right' }];

  sizes.forEach((size) => {
    it(`should render styles for size ${size}`, () => {
      const wrapper = shallow(<Body size={size} theme={theme} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  colors.forEach((color) => {
    it(`should render styles for color ${color}`, () => {
      const wrapper = shallow(<Body size="lg" theme={theme} color={color} />);
      expect(wrapper).toHaveStyleRule('color', color && getColor(color)({ theme }));
    });
  });

  alignments.forEach((alignment) => {
    it(`should render styles for alignment "${alignment}"`, () => {
      const wrapper = shallow(<Body align={alignment} theme={theme} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
