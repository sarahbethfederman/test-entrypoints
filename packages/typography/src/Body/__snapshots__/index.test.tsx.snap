// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body should render styles for alignment "[object Object]" 1`] = `
.c0 {
  margin-top: 0;
  margin-bottom: 0;
  color: #474747;
  font-size: 1em;
  font-family: 'Open Sans',sans-serif;
}

@media (min-width:0) {
  .c0 {
    text-align: left;
  }
}

@media (min-width:75.0625em) {
  .c0 {
    text-align: right;
  }
}

<p
  className="c0"
/>
`;

exports[`Body should render styles for alignment "center" 1`] = `
.c0 {
  margin-top: 0;
  margin-bottom: 0;
  color: #474747;
  text-align: center;
  font-size: 1em;
  font-family: 'Open Sans',sans-serif;
}

<p
  className="c0"
/>
`;

exports[`Body should render styles for alignment "left" 1`] = `
.c0 {
  margin-top: 0;
  margin-bottom: 0;
  color: #474747;
  text-align: left;
  font-size: 1em;
  font-family: 'Open Sans',sans-serif;
}

<p
  className="c0"
/>
`;

exports[`Body should render styles for alignment "right" 1`] = `
.c0 {
  margin-top: 0;
  margin-bottom: 0;
  color: #474747;
  text-align: right;
  font-size: 1em;
  font-family: 'Open Sans',sans-serif;
}

<p
  className="c0"
/>
`;

exports[`Body should render styles for alignment "undefined" 1`] = `
.c0 {
  margin-top: 0;
  margin-bottom: 0;
  color: #474747;
  font-size: 1em;
  font-family: 'Open Sans',sans-serif;
}

<p
  className="c0"
/>
`;

exports[`Body should render styles for size lg 1`] = `
.c0 {
  margin-top: 0;
  margin-bottom: 0;
  color: #474747;
  font-size: 1.125rem;
  line-height: calc(24 / 18);
  font-family: 'Open Sans',sans-serif;
}

<p
  className="c0"
  size="lg"
/>
`;

exports[`Body should render styles for size md 1`] = `
.c0 {
  margin-top: 0;
  margin-bottom: 0;
  color: #474747;
  font-size: 1rem;
  line-height: calc(24 / 16);
  font-family: 'Open Sans',sans-serif;
}

<p
  className="c0"
  size="md"
/>
`;

exports[`Body should render styles for size sm 1`] = `
.c0 {
  margin-top: 0;
  margin-bottom: 0;
  color: #474747;
  font-size: 0.875rem;
  line-height: calc(20 / 14);
  font-family: 'Open Sans',sans-serif;
}

<p
  className="c0"
  size="sm"
/>
`;

exports[`Body should render styles for size undefined 1`] = `
.c0 {
  margin-top: 0;
  margin-bottom: 0;
  color: #474747;
  font-size: 1em;
  font-family: 'Open Sans',sans-serif;
}

<p
  className="c0"
/>
`;
